rotkehlchen/tests/unit/solana_decoders/test_jupiter.py

```diff
@@ -152,6 +153,168 @@ def test_rfq_swap(
+        notes=f'Spend {fee_amount} SOL as transaction fee',
+        counterparty=CPT_GAS,
+    ), SolanaSwapEvent(
+        tx_ref=signature,
+        sequence_index=1,
+        timestamp=timestamp,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.SPEND,
+        asset=Asset('solana/token:EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v'),
+        amount=FVal(spend_amount := '1.99800'),
+        location_label=user_address,
+        notes=f'Swap {spend_amount} USDC in Jupiter',
+        counterparty=CPT_JUPITER,
+        address=SolanaAddress('6n9VhCwQ7EwK6NqFDjnHPzEk6wZdRBTfh43RFgHQWHuQ'),
+    ), SolanaSwapEvent(
+        tx_ref=signature,
+        sequence_index=2,
+        timestamp=timestamp,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.RECEIVE,
+        asset=Asset('solana/token:CFJxqK6Wo6CqCwa9RDwvLGUDUz3HYQXT15Lftqnupump'),
+        amount=FVal(lum_receive_amount := '66980.582126'),
+        location_label=user_address,
+        notes=f'Receive {lum_receive_amount} Luminaries as the result of a swap in Jupiter',
+        counterparty=CPT_JUPITER,
+        address=SolanaAddress('6n9VhCwQ7EwK6NqFDjnHPzEk6wZdRBTfh43RFgHQWHuQ'),
+    ), SolanaSwapEvent(
+        tx_ref=signature,
+        sequence_index=3,
+        timestamp=timestamp,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.FEE,
+        asset=Asset('solana/token:EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v'),
+        amount=FVal(trade_fee_amount := '0.002'),
+        location_label=user_address,
+        notes=f'Spend {trade_fee_amount} USDC as Jupiter platform fee',
+        counterparty=CPT_JUPITER,
+        address=SolanaAddress('6n9VhCwQ7EwK6NqFDjnHPzEk6wZdRBTfh43RFgHQWHuQ'),
+    ), SolanaEvent(
+        tx_ref=signature,
+        sequence_index=4,
+        timestamp=timestamp,
+        event_type=HistoryEventType.RECEIVE,
+        event_subtype=HistoryEventSubType.NONE,
+        asset=Asset('solana/token:So11111111111111111111111111111111111111112'),
+        amount=FVal(wsol_receive_amount := '0.000128863'),
+        location_label=user_address,
+        notes=f'Receive {wsol_receive_amount} SOL due to positive slippage in a Jupiter swap',
+        counterparty=CPT_JUPITER,
+        address=SolanaAddress('6n9VhCwQ7EwK6NqFDjnHPzEk6wZdRBTfh43RFgHQWHuQ'),
+    ), SolanaEvent(
+        tx_ref=signature,
+        sequence_index=5,
+        timestamp=timestamp,
+        event_type=HistoryEventType.SPEND,
+        event_subtype=HistoryEventSubType.FEE,
+        asset=Asset('solana/token:So11111111111111111111111111111111111111112'),
+        amount=FVal(pump_fee_amount1 := '0.000096195'),
+        location_label=user_address,
+        notes=f'Spend {pump_fee_amount1} WSOL as Pump.fun protocol fee',
+        counterparty=CPT_PUMP_FUN,
+        address=SolanaAddress('7VtfL8fvgNfhz17qKRMjzQEXgbdpnHHHQRh54R9jP2RJ'),
+    ), SolanaEvent(
+        tx_ref=signature,
```

Comment:
So do these events represent the actual asset changes of the user? That's what matters. If I process the events will I get the final balance correctly or will I be missing stuff?